.github/workflows/build-cloud-img.yml

```diff
@@ -6,10 +6,10 @@ on:
     - main
     tags:
       - v*
-  # we will NOT push the image on pull requests, only test buildability.
-  pull_request:
-    branches:
-    - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
imho it would be better to use
```
    group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```
and not cancel the main branch workflows, to prevent the red X